### PR TITLE
WIP: fix(ssr): render the teleport on the client

### LINKED ISSUE #4394: [Bug Report] In nuxt3, el-dropdown will render messy and conflict with el-popover
### Element Plus version
1.2.0-beta.3

### OS/Browsers version
windows

### Vue version
3.2.21

### Reproduction Link
https://codesandbox.io/s/quizzical-ritchie-9u4n7?file=/pages/index.vue

### Steps to reproduce
In nuxt3, el-dropdown will render messy and conflict with el-popover

### What is Expected?
Hope the rendering is normal

### What is actually happening?
Hope the rendering is normal

### LINKED ISSUE #6233: [Bug Report] el-dialog not working in nuxt3
### Element Plus version
2.0.2

### OS/Browsers version
Irrelevant

### Vue version
3.2.31

### Reproduction Link
https://stackblitz.com/edit/nuxt-starter-q7pamn?file=components/Navbar.vue

### Steps to reproduce
Click Open Dialog Button

### What is Expected?
To Open Dialog on screen

### What is actually happening?
nothing. i saw it add's `.el-popup-parent--hidden` to body but no Dialog is render

### LINKED ISSUE #7366: [Bug Report] [Component] [select, tooltip] el-select cause Hydration completed but contains mismatches error
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.33`
- Element Plus Version: `2.1.10`
- Browser / OS: `User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.75 Safari/537.36`
- Build Tool: `Vite`

### Related Component

- `el-select`
- `el-tooltip`

### Reproduction Link

[Github Repo](https://github.com/annndddy/test)

### Steps to reproduce

Create a new nuxt 3 project
add element plus @ latest
add el-select to app.vue
yarn run dev
Open browser and go to the local page

### What is Expected?

No Hydration completed but contains mismatches error

### What is actually happening?

Hydration completed but contains mismatches error occur

### Additional comments

I think any element with a el-popper cause this problem
I had test el-tooltip having the same problem

I think this is not a nuxt 3 problem because this problem occur when I was using custom entry-server to create the ssr server

### LINKED ISSUE #7372: [Bug Report] [Component] [drawer] el-drawer cannot be opened in the SSR page
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.33`
- Element Plus Version: `2.1.10`
- Browser / OS: ` Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36`
- Build Tool: `Nuxt`

### Related Component

- `el-drawer`

### Reproduction Link

[Link](https://stackblitz.com/edit/nuxt-starter-tx8wmz?file=pages%2Findex.vue)

### Steps to reproduce

click the button of openDrawer

### What is Expected?

el-drawer can be opened

### What is actually happening?

el-drawer cannot be opened

### Additional comments

open SSR in Nuxt3

error:
image

### LINKED ISSUE #7921: [Bug Report] [Component] [notification, message] Adding multiple immediate consecutive notification and message without delay will result in overlapping or messages/notifications
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.36`
- Element Plus Version: `2.2.2`
- Browser / OS: `Version 101.0.4951.64 (Official Build) (64-bit)`
- Build Tool: `Nuxt`

### Related Component

- `el-notification`
- `el-message`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uLCBvbk1vdW50ZWQgfSBmcm9tICd2dWUnXG5pbXBvcnQgeyBFbE5vdGlmaWNhdGlvbiwgRWxNZXNzYWdlIH0gZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5vbk1vdW50ZWQoKCkgPT4ge1xuICAgIEVsTm90aWZpY2F0aW9uKHtcbiAgICB0aXRsZTogJ1N1Y2Nlc3MnLFxuICAgIG1lc3NhZ2U6ICdUaGlzIGlzIGEgc3VjY2VzcyBtZXNzYWdlJyxcbiAgICB0eXBlOiAnc3VjY2VzcycsXG4gIH0pXG4gIEVsTm90aWZpY2F0aW9uKHtcbiAgICB0aXRsZTogJ1N1Y2Nlc3MnLFxuICAgIG1lc3NhZ2U6ICdUaGlzIGlzIGEgc3VjY2VzcyBtZXNzYWdlJyxcbiAgICB0eXBlOiAnc3VjY2VzcycsXG4gIH0pXG4gIHNldFRpbWVvdXQoKCkgPT4ge1xuICBcdEVsTWVzc2FnZSgndGhpcyBpcyBtZXNzYWdlIDEuJylcblx0XHRFbE1lc3NhZ2UoJ3RoaXMgaXMgbWVzc2FnZSAyLicpICBcbiAgfSwgNTAwMClcbn0pXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuXG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJfbyI6e319)

### Steps to reproduce

Trigger multiple messages or notifications continously.

### What is Expected?

notifications and messages stack up normally. All messages can be read clearly.

### What is actually happening?

messages and notifications overlap.

### Additional comments

_(empty)_

### LINKED ISSUE #7984: [Bug Report] [Component] [dropdown, select] 在nuxt3中使用使用多个el-dropdwon或者el-select，下拉的部分会一起出现并重叠在一起
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.33`
- Element Plus Version: `2.2.2`
- Browser / OS: `101.0.4951.64（正式版本） (arm64)/macos 12.3.1`
- Build Tool: `Nuxt`

### Related Component

- `el-dropdown`
- `el-select`

### Reproduction Link

[Github Repo](https://github.com/yezipi/nuxt3-elemenu-plus-bug.git)

### Steps to reproduce

```vue
<script setup lang="ts">
  import { ElDropdown, ElDropdownItem, ElDropdownMenu, ElIcon, ElSelect, ElOption } from 'element-plus'
  import { ArrowDown } from '@element-plus/icons-vue'
  
  const value = ref('')

  const options1 = [
    {
      value: 'Select Option1',
      label: 'Select Option1',
    },
    {
      value: 'Select Option2',
      label: 'Select Option2',
    },
  ]
  const options2 = [
    {
      value: 'Select BBBBB',
      label: 'Select BBBBB',
    },
    {
      value: 'Select CCCCC',
      label: 'Select CCCCC',
    },
  ]
</script>

<template>
  <div class="element-plus-demo" style="padding: 100px">
      <div>
        <h1>Dropdown 组件：</h1>
        <el-dropdown>
        <div class="el-dropdown-link">
          first Dropdown
          <el-icon class="el-icon--right">
            <arrow-down />
          </el-icon>
        </div>
        <template #dropdown>
          <el-dropdown-menu>
            <el-dropdown-item style="color: green">Dropdown first 1</el-dropdown-item>
            <el-dropdown-item style="color: green">Dropdown first 2</el-dropdown-item>
            <el-dropdown-item style="color: green">Dropdown first 3</el-dropdown-item>
          </el-dropdown-menu>
        </template>
      </el-dropdown>


      <el-dropdown>
        <div class="el-dropdown-link" style="margin-left: 100px">
          second Dropdown
          <el-icon class="el-icon--right">
            <arrow-down />
          </el-icon>
        </div>
        <template #dropdown>
          <el-dropdown-menu>
            <el-dropdown-item style="color: red">Dropdown second 1</el-dropdown-item>
            <el-dropdown-item style="color: red">Dropdown second 2</el-dropdown-item>
            <el-dropdown-item style="color: red">Dropdown second 3</el-dropdown-item>
          </el-dropdown-menu>
        </template>
      </el-dropdown>
    </div>


    <div>
      <h1>Select 组件：</h1>
      <el-select v-model="value" class="m-2" placeholder="Select" size="large">
        <el-option
          v-for="item in options1"
          :key="item.value"
          :label="item.label"
          :value="item.value"
        />
      </el-select>
      <el-select v-model="value" class="m-2" placeholder="Select">
        <el-option
          v-for="item in options2"
          :key="item.value"
          :label="item.label"
          :value="item.value"
        />
      </el-select>
    </div>

    <p style="margin-top:100px;color: #999">可以看到，下拉部分全部重叠了</p>

  </div>
</template>

<style scoped>
.example-showcase .el-dropdown-link {
  cursor: pointer;
  color: var(--el-color-primary);
  display: flex;
  align-items: center;
}
</style>

```

### What is Expected?

下拉部分不重叠

### What is actually happening?

dropdown和select的下拉部分全部重叠了

### Additional comments

_(empty)_

### LINKED ISSUE #8286: [Bug Report] [Component] [popover, popper] The Popover component isn't being rendered when setting teleported to false on server-side rendering
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.37`
- Element Plus Version: `2.2.5`
- Browser / OS: `Version 1.37.111 Chromium: 100.0.4896.79 (Official Build) (arm64) / macOS 12.3.1`
- Build Tool: `Vite`

### Related Component

- `el-popover`
- `el-popper`

### Reproduction Link

[Github Repo](https://github.com/Zeng95/vue-airbnb-ssr/tree/bug/element-plus-popover)

### Steps to reproduce

```
git clone https://github.com/Zeng95/vue-airbnb-ssr.git
cd vue-airbnb-ssr
git checkout bug/element-plus-popover
yarn install
yarn dev:ssr
open http://localhost:9000
click the "Click me" button on the home page, nothing shows up
```

### What is Expected?

The Popover component works

### What is actually happening?

The Popover component doesn't work

### Additional comments

_(empty)_

### LINKED ISSUE #9414: [Component] NUXT3中含有el-tooltip所有组件服务器渲染异常
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.37`
- Element Plus Version: `2.2.14`
- Browser / OS: `Microsoft Edge 104.0.1293.63`
- Build Tool: `Nuxt`

### Related Component

- `el-select`
- `el-tooltip`
- `el-select-v2`
- `el-tooltip-v2`

### Reproduction Link

[Github Repo](https://github.com/Felix19901229/element-plus-bug.git)

### Steps to reproduce

两个以上ElSelect数据会被合并成一个弹窗显示。
两个以上ElTooltip,只有用户最先选中的组件正常工作，其他均报错失效。

### What is Expected?

两个以上ElSelect的弹窗独立显示，数据相互独立。
两个以上ElTooltip均能独立正常工作。

### What is actually happening?

两个以上ElSelect,弹窗数据被合并。
两个以上ElTooltip,只有一个工作。

### Additional comments

_(empty)_

### LINKED ISSUE #9418: [Build] [All] Nuxt hydration failed (component - Menu): Hydration children mismatch in <div>: server rendered element contains fewer child nodes than client vdom
Bug Type: **`Build`**

## Environment

- Vue Version: `3.2.37`
- Element Plus Version: `2.2.14`
- Browser / OS: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:104.0) Gecko/20100101 Firefox/104.0`
- Build Tool: `Nuxt`

### Related Component

- `All`

### Reproduction Link

[Docs](https://element-plus.org/en-US/component/menu.html)

### Steps to reproduce

just use menu component in nuxt, without client-only

### What is Expected?

I want to nuxt rightly hydrated menu component (with submenu)

### What is actually happening?

I've got a error:
[Vue warn]: Hydration children mismatch in <div>: server rendered element contains fewer child nodes than client vdom. 
  at <ElTooltipContent ref="contentRef" aria-label=undefined boundaries-padding=0  ... > 
  at <ElPopperRoot ref="popperRef" role="tooltip" > 
  at <ElTooltip ref=Ref< null > visible=false effect="light"  ... > 
  at <ElSubMenu index="1" > 

### Additional comments

_(empty)_

### LINKED ISSUE #9538: [Component] [menu, dropdown] menu和dropdown 同时使用在服务端渲染疑似有问题 / There seems to be a problem with server-side rendering when both menu and dropdown are used at the same time
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.37`
- Element Plus Version: `2.2.15`
- Browser / OS: `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.0.0 Safari/537.36`
- Build Tool: `Vite`

### Related Component

- `el-menu`
- `el-dropdown`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcbmNvbnN0IGFjdGl2ZUluZGV4ID0gcmVmKCcxJylcbmNvbnN0IGFjdGl2ZUluZGV4MiA9IHJlZignMScpXG5jb25zdCBoYW5kbGVTZWxlY3QgPSAoa2V5OiBzdHJpbmcsIGtleVBhdGg6IHN0cmluZ1tdKSA9PiB7XG4gICAgY29uc29sZS5sb2coa2V5LCBrZXlQYXRoKVxufVxuPC9zY3JpcHQ+XG48dGVtcGxhdGU+XG4gICAgPGVsLWRyb3Bkb3duPlxuICAgICAgICA8c3BhbiBjbGFzcz1cImVsLWRyb3Bkb3duLWxpbmtcIj5cbiAgICAgICAgICAgIERyb3Bkb3duIExpc3RcbiAgICAgICAgICAgIDxlbC1pY29uIGNsYXNzPVwiZWwtaWNvbi0tcmlnaHRcIj5cbiAgICAgICAgICAgICAgICA8YXJyb3ctZG93biAvPlxuICAgICAgICAgICAgPC9lbC1pY29uPlxuICAgICAgICA8L3NwYW4+XG4gICAgICAgIDx0ZW1wbGF0ZSAjZHJvcGRvd24+XG4gICAgICAgICAgICA8ZWwtZHJvcGRvd24tbWVudT5cbiAgICAgICAgICAgICAgICA8ZWwtZHJvcGRvd24taXRlbT5BY3Rpb24gMTwvZWwtZHJvcGRvd24taXRlbT5cbiAgICAgICAgICAgICAgICA8ZWwtZHJvcGRvd24taXRlbT5BY3Rpb24gMjwvZWwtZHJvcGRvd24taXRlbT5cbiAgICAgICAgICAgICAgICA8ZWwtZHJvcGRvd24taXRlbT5BY3Rpb24gMzwvZWwtZHJvcGRvd24taXRlbT5cbiAgICAgICAgICAgICAgICA8ZWwtZHJvcGRvd24taXRlbSBkaXNhYmxlZD5BY3Rpb24gNDwvZWwtZHJvcGRvd24taXRlbT5cbiAgICAgICAgICAgICAgICA8ZWwtZHJvcGRvd24taXRlbSBkaXZpZGVkPkFjdGlvbiA1PC9lbC1kcm9wZG93bi1pdGVtPlxuICAgICAgICAgICAgPC9lbC1kcm9wZG93bi1tZW51PlxuICAgICAgICA8L3RlbXBsYXRlPlxuICAgIDwvZWwtZHJvcGRvd24+XG4gICAgPGVsLW1lbnUgOmRlZmF1bHQtYWN0aXZlPVwiYWN0aXZlSW5kZXhcIiBjbGFzcz1cImVsLW1lbnUtZGVtb1wiIG1vZGU9XCJob3Jpem9udGFsXCIgQHNlbGVjdD1cImhhbmRsZVNlbGVjdFwiPlxuICAgICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiMVwiPlByb2Nlc3NpbmcgQ2VudGVyPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1zdWItbWVudSBpbmRleD1cIjJcIj5cbiAgICAgICAgICAgIDx0ZW1wbGF0ZSAjdGl0bGU+V29ya3NwYWNlPC90ZW1wbGF0ZT5cbiAgICAgICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIyLTFcIj5pdGVtIG9uZTwvZWwtbWVudS1pdGVtPlxuICAgICAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjItMlwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiMi0zXCI+aXRlbSB0aHJlZTwvZWwtbWVudS1pdGVtPlxuICAgICAgICAgICAgPGVsLXN1Yi1tZW51IGluZGV4PVwiMi00XCI+XG4gICAgICAgICAgICAgICAgPHRlbXBsYXRlICN0aXRsZT5pdGVtIGZvdXI8L3RlbXBsYXRlPlxuICAgICAgICAgICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIyLTQtMVwiPml0ZW0gb25lPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgICAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjItNC0yXCI+aXRlbSB0d288L2VsLW1lbnUtaXRlbT5cbiAgICAgICAgICAgICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiMi00LTNcIj5pdGVtIHRocmVlPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgICAgICA8L2VsLXN1Yi1tZW51PlxuICAgICAgICA8L2VsLXN1Yi1tZW51PlxuICAgICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiM1wiIGRpc2FibGVkPkluZm88L2VsLW1lbnUtaXRlbT5cbiAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjRcIj5PcmRlcnM8L2VsLW1lbnUtaXRlbT5cbiAgICA8L2VsLW1lbnU+XG48L3RlbXBsYXRlPlxuXG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)

### Steps to reproduce

将Element Plus Playground的内容转为服务端渲染即可
---------------
Convert the content of Element Plus Playground to server-side rendering

### What is Expected?

以上组件在 Element Plus Playground 客户端渲染时没有问题，如果使用SSR渲染则会出现2个菜单内容会出现在一起
---------
The above components have no problem when rendered on the Element Plus Playground client side. If they are rendered using SSR, 2 menu contents will appear together.

### What is actually happening?

与客户端渲染表现一致
--------------
Consistent with client-side rendering performance

### Additional comments

这是使用服务器渲染浏览器所获取的代码
The html code returned by SSR rendering is as follows
```
<!DOCTYPE html>
<html lang="en">
    <head>
        <script type="text/javascript" nonce="3f09ebc05ed24f08bed427df0c6" src="//local.adguard.org?ts=1661820979529&amp;type=content-script&amp;dmn=localhost:3000&amp;pth=%2Fhello&amp;app=chrome.exe&amp;css=3&amp;js=1&amp;rel=1&amp;rji=1&amp;sbe=1"></script>
        <script type="text/javascript" nonce="3f09ebc05ed24f08bed427df0c6" src="//local.adguard.org?ts=1661820979529&amp;name=Search%20By%20Image&amp;name=AdGuard%20Assistant&amp;name=AdGuard%20Extra&amp;name=AdGuard%20Popup%20Blocker&amp;type=user-script"></script>
        <script type="module" src="/@vite/client"></script>
        <meta charset="UTF-8"/>
        <link rel="icon" type="image/svg+xml" href="/vite.svg"/>
        <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
        <title>Vite + Vue + TS</title>
    </head>
    <body>
        <div id="app" style="max-width: 1000px;margin:0 auto">
            <!--[-->
            <div class="el-dropdown">
                <!--[-->
                <span class="el-dropdown-link el-tooltip__trigger el-tooltip__trigger" id="el-id-4984-0" role="button" tabindex="0" aria-controls="el-id-4984-1" aria-expanded="false" aria-haspopup="menu">
                    Dropdown List 
                    <i class="el-icon el-icon--right" style="">
                        <!--[-->
                        <arrow-down></arrow-down>
                        <!--]-->
                    </i>
                </span>
                <!--teleport start-->
                <!--teleport end-->
                <!--]-->
                <!--v-if-->
            </div>
            <ul role="menubar" style="--el-menu-level:0;" class="el-menu el-menu--horizontal el-menu-demo">
                <li class="el-menu-item is-active" role="menuitem" tabindex="-1">
                <!--[-->
                <!--[-->
                Processing Center
                <!--]-->
                <!--[-->
                <!--]-->
                <!--]-->
                </li>
                <li class="el-sub-menu" role="menuitem" ariahaspopup="true" ariaexpanded="false">
                    <!--[-->
                    <div class="el-sub-menu__title el-tooltip__trigger el-tooltip__trigger" style="border-bottom-color:transparent;color:;background-color:;">
                        <!--[-->
                        Workspace
                        <!--]-->
                        <i class="el-icon el-sub-menu__icon-arrow" style="transform:none;">
                            <!--[-->
                            <svg viewbox="0 0 1024 1024" xmlns="http://www.w3.org/2000/svg">
                                <path fill="currentColor" d="M831.872 340.864 512 652.672 192.128 340.864a30.592 30.592 0 0 0-42.752 0 29.12 29.12 0 0 0 0 41.6L489.664 714.24a32 32 0 0 0 44.672 0l340.288-331.712a29.12 29.12 0 0 0 0-41.728 30.592 30.592 0 0 0-42.752 0z"></path>
                            </svg>
                            <!--]-->
                        </i>
                    </div>
                    <!--teleport start-->
                    <!--teleport end-->
                    <!--]-->
                </li>
                <li class="el-menu-item is-disabled" role="menuitem" tabindex="-1">
                <!--[-->
                <!--[-->
                Info
                <!--]-->
                <!--[-->
                <!--]-->
                <!--]-->
                </li>
                <li class="el-menu-item" role="menuitem" tabindex="-1">
                <!--[-->
                <!--[-->
                Orders
                <!--]-->
                <!--[-->
                <!--]-->
                <!--]-->
                </li>
            </ul>
            <!--]-->
        </div>
        <!-- <script type="module" src="/src/main.ts"></script> -->
        <script type="module" src="/src/entry-client.js?t=1661827370682"></script>
    </body>
</html>
```

### PR DESCRIPTION
**Pull request by tolking**
close #4394, close #6233, close #7366, close #7372, close #7921, close #7984, close #8286, close #9414, close #9418, close #9538

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

Reference documentation [Vue](https://vuejs.org/guide/scaling-up/ssr.html#teleport), [Nuxt](https://v3.nuxtjs.org/guide/features/teleports)

### CODE REVIEW

File: packages/components/dialog/src/dialog.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <teleport to="body" :disabled="!appendToBody">
+  <teleport v-if="isClient" to="body" :disabled="!appendToBody">
```
Comment (btea) [inline packages/components/dialog/src/dialog.vue:2]: According to the nuxt documentation, it seems to support teleport to `body`.

Comment (tolking) [reply packages/components/dialog/src/dialog.vue:2]: It may be necessary to verify that native(Vue SSR) also supports

Comment (tolking) [reply packages/components/dialog/src/dialog.vue:2]: If using the native SSR to render multiple content `teleport` to the `body`, the content will be displayed in a DOM.

Comment (btea) [reply packages/components/dialog/src/dialog.vue:2]: OK, I see

**Review by btea — APPROVED** [review]

**Comment by holazz** [comment]
Why not import `isClient` directly from vueuse?  🤔
https://github.com/vueuse/vueuse/blob/2f81b30ac5710b511999fdfd33543cee68d47713/packages/shared/utils/is.ts#L1

**Comment by tolking** [comment]
> Why not import `isClient` directly from vueuse? 🤔 https://github.com/vueuse/vueuse/blob/2f81b30ac5710b511999fdfd33543cee68d47713/packages/shared/utils/is.ts#L1

The two are completely different, the `teleport` component must be rendered in the onMounted.

**Comment by holazz** [comment]
> > Why not import `isClient` directly from vueuse? 🤔 https://github.com/vueuse/vueuse/blob/2f81b30ac5710b511999fdfd33543cee68d47713/packages/shared/utils/is.ts#L1
> 
> 
> 
> The two are completely different, the `teleport` component must be rendered in the onMounted.


Ok, got it.

**Review by holazz — APPROVED** [review]

**Comment by warflash** [comment]
Hey, quick question: Conditionally rendering hydration mismatches only on the client does work, but doesn't that also sort of defeat most of the value one gets from server side rendered pages? Crawlers won't be able to pick up any content of client only components on first render and will only be able to index it on full renders which are way less frequent. 
Just wondering whether in theory there is a proper ssr solution that'd be possible or is this the only way?

**Comment by tolking** [comment]
> Hey, quick question: Conditionally rendering hydration mismatches only on the client does work, but doesn't that also sort of defeat most of the value one gets from server side rendered pages? Crawlers won't be able to pick up any content of client only components on first render and will only be able to index it on full renders which are way less frequent. Just wondering whether in theory there is a proper ssr solution that'd be possible or is this the only way?

If you implement SSR by yourself, you can rewrite the render function implementation. but I have not found that Nuxt provides a similar interface.

Currently Nuxt only support the `<Teleport to="body">`, we can also wait for Nuxt to fully support.

**Review by holazz — APPROVED** [review]

**Pull request closed without merging**